docs/spec/spec.md

```diff
@@ -3647,3 +3749,141 @@ isolated function followersLoaderFunction(readonly & anydata[] ids) returns Foll
+```
+
+###### Example: Operation-level Cache Enabling and Eviction
+
+```ballerina
+import ballerina/graphql;
+
+@graphql:ServiceConfig {
+    cacheConfig: {
+        enabled: true,
+        maxAge: 50
+    }
+}
+service /graphql on new graphql:Listener(9090) {
+    private string name = "Ballerina GraphQL";
+    private string 'type = "code first";
+
+    resource function get name() returns string {
+        return self.name;
+    }
+
+    resource function get 'type() returns string {
+        return self.'type;
+    }
+
+    remote function updateName(graphql:Context context, string name) returns string|error {
+        check context.invalidate("name");
+        self.name = name
+        return self.name;
+    }
+}
+```
+
+In this example, caching is enabled at the operation level. Therefore, the field `name` and `type` will be cached. When updating the name with a mutation, the cached values become invalid. Hence, the `invalidate` function can be used to evict the existing cache values.
+
+###### Example: Field-level Cache Enabling and Eviction
+
+```ballerina
+import ballerina/graphql;
+
+type Friend record {|
+    readonly string name;
+    int age;
+    boolean isMarried;
+|};
+
+service /graphql on new graphql:Listener(9090) {
+    private table<Friend> key(name) friends = table [
+        {name: "Skyler", age: 45, isMarried: true},
+        {name: "Jesse Pinkman", age: 23, isMarried: false}
+    ];
+
+    isolated resource function get friends(boolean isMarried = false) returns Person[] {
+        if isMarried {
+            return from Friend friend in self.friends
+                where friend.isMarried == true
+                select new Person(friend.name, friend.age, isMarried);
+        }
+        return from Friend friend in self.friends
+            where friend.isMarried == false
+            select new Person(friend.name, friend.age, isMarried);
+    }
+
+    isolated remote function updateAge(graphql:Context context, string name, int age) returns Person|error {
+        check context.invalidate("friends.age");
+        Friend friend = self.friends.get(name);
+        self.friends.put({name: name, age: age, isMarried: friend.isMarried});
+        return new Person(name, age, friend.isMarried);
+    }
+}
+
+public isolated distinct service class Person {
+    private final string name;
+    private final int age;
+    private final boolean isMarried;
+
+    public isolated function init(string name, int age, boolean isMarried) {
+        self.name = name;
+        self.age = age;
+        self.isMarried = isMarried;
+    }
+
+    isolated resource function get name() returns string {
+        return self.name;
+    }
+
+    @graphql:ResourceConfig {
+        cacheConfig: {
+            enabled: true,
+            maxAge 20
+        }
+    }
+    isolated resource function get age() returns int {
+        return self.age;
+    }
+
+    isolated resource function get isMarried() returns boolean {
```

Comment:
Maybe we can use another cache config here to disable caching for this field. (Let's use comments to describe the behavior as well).